pkg/backend/openshiftcluster.go

```diff
@@ -203,3 +225,31 @@ func (ocb *openShiftClusterBackend) endLease(stop func(), doc *api.OpenShiftClus
 	_, err = ocb.db.OpenShiftClusters.EndLease(doc.Key, provisioningState, failedProvisioningState)
 	return err
 }
+
+// handleDelete handles document state during context cancelation and backend
+// worker shutdown. If backend worker context is canceled, it will mark document
+// as Failed. We need to wait for it and update to Deleting.
+// End to end state change flow should be:
+// Creating -> Deleting -> Failed -> Deleting
```

Comment:
it would be much better if you can prevent the `Failed` write to the DB in the first place.